packages/langium/src/utils/promise-utils.ts

```diff
@@ -0,0 +1,34 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+export type MaybePromise<T> = T | Promise<T>
+
+/**
+ * Delays the execution of the current code to the next tick of the event loop.
+ */
+export function delayNextTick(): Promise<void> {
+    return new Promise(resolve => {
+        setImmediate(resolve);
+    });
+}
+
+let lastTick = -1;
```

Comment:
I can see that leading to a lot of interruptions of the event loop, since they each have their own timer. Instead, what this does is that an interrupt will be triggered every 5 ms on a global level. I don't think we need higher fidelity, do we? Even if multiple callers call this method, it will only interrupt the event loop once, which is exactly what we want.